modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/item/selector/FDSViewItemSelector.tsx

```diff
@@ -76,7 +86,11 @@ const FDSViewItemSelector = ({
 
 						<ClayButton
 							className="item-preview selector-button"
-							data-value={`{"classPK": "${selectedId}", "className": "${className}",  "classNameId": "${classNameId}"}`}
+							data-value={`{
+								"className": "${className}", 
+								"classNameId": "${classNameId}", 
+								"classPK": "${selectedId}", 
+								"externalReferenceCode": "${externalReferenceCode}"}`}
```

Comment:
![](https://media4.giphy.com/media/sG4PBWRjI4GSVCDXEq/200.gif)


I know this code was already there, but is there a better way to pass this data without adding it to the DOM like this?

Reply:
I'm curious about how this pattern works under the hood

Reply:
After a closer look, this seems related to how selection modal [grabs the selected item](https://github.com/liferay/liferay-portal/blob/b50e8cc6c998a8b868439b789a95d5fe5a93e171/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js#L622-L626) data